### PR TITLE
docs: adapt doc for new attachment logic

### PR DESCRIPTION
**Pull request by Dam-Buty**

### CODE REVIEW

File: typescript-client/api-reference/api-reference.mdx [hunk 1]
```diff
@@ -66,7 +66,61 @@ The unique identifier of the step to delete.
...
 A `Promise` that resolves to the ID of the deleted step.
 
-### uploadFile()
+### createAttachment()
+
+> **createAttachment**(`params`): `Promise`\<`object`\>
+
+Uploads a file to Literal AI, and creates an attachment out of it. If called within a step wrapper, the attachment will be automatically added to the step.
+
+This method supports uploading either through direct content or via a file path. It first signs the upload through a pre-configured endpoint and then proceeds to upload the file using the signed URL.
```
Comment (desaxce) [inline typescript-client/api-reference/api-reference.mdx:75]: I didn't understand the signing part of the URL. I can see that it means we're secure.

File: typescript-client/api-reference/api-reference.mdx [hunk 2]
```diff
@@ -66,7 +66,61 @@ The unique identifier of the step to delete.
...
+
+#### Parameters
+
+▪ **params**: `object`
+
+The parameters for uploading a file, including:
```
Comment (desaxce) [inline typescript-client/api-reference/api-reference.mdx:81]: Not the purpose for this PR, but ideally users see a summary JSON object like this:
```
{
    content: ReadableStream | ReadStream ...,
    path: Maybe<string>
}
```

Either that or some clearer indentation.

Comment (desaxce) [reply typescript-client/api-reference/api-reference.mdx:81]: Like that actually: 
image

File: typescript-client/development/changelog.mdx [hunk 3]
```diff
@@ -2,6 +2,19 @@
 title: Changelog
 ---
 
+# `0.0.511`
+
+### New Features
+
+This release introduces a new method to create attachments : `client.api.createAttachment`. This method is a wrapper around the `uploadFile` method and the `new Attachment` constructor. It allows you to create an attachment and attach it to a step in a single call.
```
Comment (desaxce) [inline typescript-client/development/changelog.mdx:9]: I would add the keyword `context-aware` method as you mentioned in the code, as in "we handle it for you, we make your life easier".

**Review by desaxce — APPROVED** [review]

File: typescript-client/api-reference/api-reference.mdx (hunk 1 above)
Comment (Dam-Buty) [reply typescript-client/api-reference/api-reference.mdx:75]: honestly it's not that important a point for the end user so i removed it and replaced it with a vague statement : "securely uploads a file to Literal AI ...".

File: typescript-client/api-reference/api-reference.mdx (hunk 2 above)
Comment (Dam-Buty) [reply typescript-client/api-reference/api-reference.mdx:81]: You're right i'm not a fan of this presentation either. As a whole this API reference feels very unsatisfying, it is just a list with no added information. I'll try to overhaul it when i find a minute.

File: typescript-client/development/changelog.mdx (hunk 3 above)
Comment (Dam-Buty) [reply typescript-client/development/changelog.mdx:9]: yes, i changed it and mentioned both wrappers and context

File: typescript-client/api-reference/api-reference.mdx (hunk 2 above)
Comment (desaxce) [reply typescript-client/api-reference/api-reference.mdx:81]: Plan is to focus on doc during August apparently.

**Review by desaxce — APPROVED** [review]

**Pull request merged**